### ISSUE TITLE
Security Concern - why is it necessary to enable kernel.unprivileged_userns_clone?

### ISSUE THREAD

**Issue by Nintorac** [input-output-hk/daedalus#2025]
During installation on Debian Buster an error comes up prompting me with the following

```
user@machine:~/mnt2/samples$ ./daedalus-1.1.0-mainnet-12849.bin 
Run the following to enable unprivileged namespace use:
sudo bash -c "sysctl -w kernel.unprivileged_userns_clone=1 ; echo kernel.unprivileged_userns_clone=1 > /etc/sysctl.d/nix-user-chroot.conf"
```

A quick google gives this [security exchange](https://security.stackexchange.com/questions/209529/what-does-enabling-kernel-unprivileged-userns-clone-do) answer which doesn't clear things up or inspire confidence.

Further googling takes me to [your installation instructions](https://iohk.zendesk.com/hc/en-us/articles/900000776446-Installing-Daedalus-1-1-0-Mainnet-on-Linux-) which gloss over these commands and don't inform me why you need them. 

Additionally it isn't clear if these settings are required permanently or can be re enabled after installation. 

So if possible could you clarify these points for me:
1. Is this a safe operation that will not open me up to kernel vulnerabilities
2. Why are these needed?
3. Can I disable them afterwards?

Thanks!

**Comment by nikolaglumac** [comment]
cc @disassembler

**Comment by Tomassmits** [comment]
I quite dislike the combination of a cryptocurrency wallet and unprivileged namespace, or am I just the paranoid one here?

**Comment by carrotshadow** [comment]
+1.
Some explanation would be useful please.
Guess I try virtualisation next. Probably build from source to choose installation directories too.
Most likely I just lack the knowledge, and documentation, but I did search.
Thanks for all efforts.

**Comment by nikolaglumac** [comment]
cc @disassembler

**Comment by outermeasure** [comment]
+1 What are the side effects of this? Why would daedalus need this?

**Comment by nikolaglumac** [comment]
cc @disassembler

**Comment by schoenenbach** [comment]
+1 maybe providing an AppImage would be a way to deploy across linux distros for use lazy people who don't want to build from source. Besides that, I second all the questions above.

**Comment by LeoniePhiline** [comment]
Is there any progress on this?

**Comment by nikolaglumac** [comment]
@disassembler @cleverca22 please help us out here 🙏

**Comment by JasonSwindle** [comment]
Also to note, not all distros support this, thus making daedalus unusable.

**Comment by hardcore-sushi** [comment]
I'd like to run daedalus inside a container. However I can't due to the unprivileged user namespace feature required by the nix installer. Is it possible to distribute a more common installer like a .deb file or directly the binary files (without installation) like an AppImage or an ELF binary ?

**Comment by Nintorac** [comment]
Happy (belated) birthday, Bug Report!!

**Comment by nikolaglumac** [comment]
Sorry for this taking so long @Nintorac 😞 
I wish I could give you the answer myself...
@cleverca22 @disassembler please help us here - it's been over a year 🙏

**Comment by paulrougieux** [comment]
Besides the error message mentioned above, enabling unprivileged user namespaces is part of the [daedalus installation instructions on this IOHK support page](https://iohk.zendesk.com/hc/en-us/articles/900000776446-Installing-Daedalus-Mainnet-on-Linux-). Can someone explain why this is necessary?

**Comment by nikolaglumac** [comment]
cc @disassembler

**Comment by paulrougieux** [comment]
@Nintorac or @nikolaglumac improving the title will help attract developers to comment on this issue. Can you please add "why is it necessary to enable kernel.unprivileged_userns_clone?" to the title?

**Comment by nikolaglumac** [comment]
Done @paulrougieux 👍

**Comment by Nintorac** [comment]
Hi, not sure if this is still a valid issue, haven't tried to install Daedalus in a while but assuming since it's still open that it's still happening.

Recently it has been discovered there is a kernel bug in all Linux kernels prior to 5.1rc1 that allows complete kernel priivelege escalation with a local account. The mitigation is to disable this setting that Daedalus requires so perhaps it is time to revisit the issue?

https://www.zdnet.com/article/nasty-linux-kernel-bug-found-and-fixed/

**Comment by danielmain** [comment]
Hi @Nintorac I have read a little about this knob and [why it was introduced](https://lwn.net/Articles/673597/) and as far I see my deduction is because of the sandbox approach nix is design. I am not sure how it is done by AppImage or Flatpak.

We have at IO many experts and committers in the Nix project and I am trying to contact them directly to get a satisfactory answer for you.

**Comment by danielmain** [comment]
Hi @Nintorac I have got this from our DevOps regarding that knob [image](https://user-images.githubusercontent.com/4303099/150867394-fad6b9ea-bce8-4c97-bd0b-88e7b9ed811c.jpeg)

So I don't think this is going to change from our side just because some distros disable it. That is why I'm going to close this issue.

**Comment by Nintorac** [comment]
Hi @danielmain thanks for chasing this up.

Still not super satisfactory answer, I'm definitely no expert so I may just be over reacting but I've never had an app request something like this before. And for an app where security is paramount it seems there should be some sort of deeper explanation and discussion around risks/benefits. Would this pass a security audit?

Additionally the vulnerability I mentioned hasn't been addressed. Are the dev team aware of that?

**Comment by danielmain** [comment]
Hi @Nintorac I can understand your security concerns if we use applications that handles "money". 

But as far I can see here in the [discussion ](https://lwn.net/Articles/673614/), that this switch was never conceived for Desktop environments according to an interview with the creator:
 
<img width="787" alt="image" src="https://user-images.githubusercontent.com/4303099/150932529-cf70bff9-059d-4a93-8de1-89a2413fdc9e.png">

So, starting from that, Daedalus is a Desktop environment, it will be used in desktop environments.

Apart from that, I can understand why is this whole “Security concern” coming, it promises something that isn't really the case.
<img width="792" alt="image" src="https://user-images.githubusercontent.com/4303099/150932822-79009e25-13dc-4555-b4fc-b8b153bd4a7a.png">


I can invite you to read the whole article https://lwn.net/Articles/673597/

**Comment by carrotshadow** [comment]
As far as I can gather, this enablement, which is required to install the Daedalus wallet on the desktop
renders that whole 'desktop system' more vulnerable to abuse by other processes.
I could not install Daedalus at all on a Linux system unless that system was built with SystemD.
When I tried to install on a problematic SystemD variant, then this compromised namespace issue became apparent.
It was for this reason that I found myself at this thread asking questions.
It was necessary for me to run Daedalus in a containerised instance of {SystemD Linux distribution} to mitigate this risk, which remains non convenient. 
For those users whom do not check the consequences of installation of packages, this can obviously be an issue.
Furthermore, Daedalus wallet installation does not provide choice for installation target directories nor filesystems; causing resource hogging issues. Bad combination.
It appears to me that this issue has been closed on confused logic.
I recognise that I could also be confused.
